CLI - auto sort nodelist in config and on `dexbot-cli run`

thehapax opened the pull request:
Add option to the CLI configure to sort node list based on latency and remove those nodes that are too far away.  Also, when starting by `dexbot-cli run`, the nodes from the config file are checked and sorted for latency before starting the bot. 

this is an initial version, no gui updates have been made. hope this will help the node issues. 

next update : switch to another node when there is too much latency in order placement.

bitfag commented on dexbot/node_manager.py:
```diff
@@ -0,0 +1,52 @@
+
+    df_nodes = pd.DataFrame(latency_info)
+    df_active = df_nodes.dropna()
+    df_sorted = df_active.sort_values('Latency', ascending=True)
```
I think this could be done using `sorted()`, also we're not using pandas in dexbot yet, there is no requirements.txt dependency on it. `dropna()` could be replaced with simple list comprehension.

thehapax replied:
replaced.

bitfag commented on dexbot/node_manager.py:
```diff
@@ -0,0 +1,52 @@
+    """
+    check latency of an individual node
+    """
+    print('#', end='', flush=True)
```
`print()` should be replaced with logger calls

thehapax replied:
replaced with logging

thehapax commented:
Note: could also use `tqdm` for progress bar, as a future feature.

joelvai commented:
One thing that came to my mind. Wouldn't it be better to have the node pinging in `dexbot-cli run / configure` as optional parameter? Let's say I want to order my nodes and then run dexbot. The bot will have to ping trough all the nodes for three times in total. Also it would make the order of the nodes in configuration file kinda pointless since it will ping all the nodes anyway when running the bot. 

Removing bad nodes automatically when sorting is nice :+1:

PermieBTS commented:
Digital Lucifer (@dls-cipher) has cleansed the node list removing unsuitable nodes.
https://github.com/Codaone/DEXBot/pull/643

thehapax commented:
> Digital Lucifer (@dls-cipher) has cleansed the node list removing unsuitable nodes.
> Relevant to this pull: #643

Reviewed, most of the nodes are reachable from my test 21/25 and @joelvai  can merge at anytime

thehapax commented:
> One thing that came to my mind. Wouldn't it be better to have the node pinging in `dexbot-cli run / configure` as optional parameter? Let's say I want to order my nodes and then run dexbot. The bot will have to ping trough all the nodes for three times in total. Also it would make the order of the nodes in configuration file kinda pointless since it will ping all the nodes anyway when running the bot.
> 
> Removing bad nodes automatically when sorting is nice 👍

@joelvai Updated and fixed above in latest commits. Let me know if you want me to update  `def measure_latency(nodes):` in  main_controller or if you want to do it.  I fixed as much pep8 and the subprocess complaint as possible.
